British Policing Script/Menus.cs

```diff
@@ -1,99 +1,86 @@
 
         public static void InitialiseMenus()
         {
+            InitializeMenuPool();
+            InitializeCourtsMenu();
 
-
-            //bigMessage = new BigMessageThread(true);
             Game.FrameRender += Process;
-            _MenuPool = new MenuPool();
-            ChecksMenu = new UIMenu("Checks", "");
-            _MenuPool.Add(ChecksMenu);            
-           
+            MainLogic();
+        }
+
+        private static void InitializeMenuPool()
+        {
+            MenuPool.Add(ChecksMenu = new UIMenu("Checks", ""));
+
             ChecksMenu.AddItem(CheckNameItem = new UIMenuItem("Name/Vehicle Records Check"));
-            
             ChecksMenu.AddItem(CheckInsuranceItem = new UIMenuItem("Insurance Checks"));
             ChecksMenu.AddItem(CheckCourtResultsItem = new UIMenuItem("Magistrates' Court Results"));
-            //ChecksMenu.SetMenuWidthOffset(0)
-            ChecksMenu.RefreshIndex();
+
             ChecksMenu.OnItemSelect += OnItemSelect;
-            //ChecksMenu.OnListChange += OnListChange;
-            //_MenuPool.ProcessMenus();
             ChecksMenu.MouseControlsEnabled = false;
             ChecksMenu.AllowCameraMovement = true;
 
-            CourtsMenu = new TabView("~b~~h~San Andreas Magistrates' Court");
-
-            
-            
-            CourtsMenu.Tabs.Add(PendingResultsList = new TabSubmenuItem("Pending Results", EmptyItems));
-            CourtsMenu.Tabs.Add(PublishedResultsList = new TabSubmenuItem("Results", EmptyItems));
-            
-            CourtsMenu.RefreshIndex();
-            MainLogic();
-            //CourtSystem.CreateNewCourtCase("Zach Houseknecht", new DateTime(1990, 2, 1), "speeding at 82MPH in a 30MPH limit", DateTime.Now.AddDays(-1), true, "Fined $500 and licence suspended for 56 days.", DateTime.Now, true);
-            //CourtSystem.CreateNewCourtCase("Jeff Favignano", new DateTime(1990, 2, 1), "Armed Robbery", DateTime.Now, true, "Sentenced to 6 years in prison", DateTime.Now.AddDays(1), false);
+            ChecksMenu.RefreshIndex();
+        }
 
+        private static void InitializeCourtsMenu()
+        {
+            CourtsMenu = new TabView("~b~~h~San Andreas Magistrates' Court");
 
+            // use "AddTab" rather than "Tabs.Add" as it will throw an exception in "TabSubmenuItem.ProcessControls"
+            // because the "Parent" won't be set in the submenu items causing a "NullPointerException"
+            CourtsMenu.AddTab(PendingResultsList = new TabSubmenuItem("Pending Results", EmptyItems));
+            CourtsMenu.AddTab(PublishedResultsList = new TabSubmenuItem("Results", EmptyItems));
 
+            CourtsMenu.RefreshIndex();
         }
+
         private static void OnItemSelect(UIMenu sender, UIMenuItem selectedItem, int index)
         {
-            if (sender == ChecksMenu)
+            if (sender != ChecksMenu)
+                return;
```

Comment:
Please make use of braces `{` `}` at all times, not doing so makes code more prone to unexpected errors.

Reply:
Updated with braces